src/Eloquent/Builder.php

```diff
@@ -229,7 +230,11 @@ public function decrement($column, $amount = 1, array $extra = [])
         return parent::decrement($column, $amount, $extra);
     }
 
-    /** @inheritdoc */
+    /**
+     * @return ($value is callable ? Collection : ($value is null ? Collection : Expression))
+     *
+     * @inheritdoc
```

Comment:
Was unsure if I should leave the `@inheritdoc` because it might override the conditional return type?